drivers/bluetooth/hci/ipc.c

```diff
@@ -239,12 +251,32 @@ static void bt_ipc_rx(const struct device *dev, const uint8_t *data, size_t len)
 
 	if (buf) {
 		LOG_DBG("Calling bt_recv(%p)", buf);
-		ipc->recv(dev, buf);
+		ipc->recv(ipc->ipc, buf);
 
 		LOG_HEXDUMP_DBG(buf->data, buf->len, "RX buf payload:");
 	}
 }
 
+static void inc_msgq_proc(void *p1, void *p2, void *p3)
+{
+	struct ipc_data *ipc = p1;
+
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	while (true) {
+		struct hci_pkt pkt;
+		int err;
+
+		k_msgq_get(&ipc->inc_msgq, &pkt, K_FOREVER);
+
+		bt_ipc_rx(ipc, pkt.data, pkt.len);
+
+		err = ipc_service_release_rx_buffer(&ipc->hci_ept, pkt.data);
+		__ASSERT(err == 0, "Failed to release rx buffer");
```

Comment:
There is currently discrepancy between icbmsg implementation of this function and documentation. Icbmsg sometimes returns POSITIVE value on successful release. This should be fixed in the future, but for now I would change this to "err >= 0".